pkg/components/yql_agent.go

```diff
@@ -141,8 +141,10 @@ func (yqla *yqlAgent) doSync(ctx context.Context, dry bool) (ComponentStatus, er
 
 	if yqla.NeedSync() {
 		if !dry {
+			token, _ := yqla.secret.GetValue(consts.TokenSecretKey)
 			ss := yqla.server.buildStatefulSet()
 			container := &ss.Spec.Template.Spec.Containers[0]
+			container.Command = []string{"sh", "-c", fmt.Sprintf("echo -n %s > /usr/yql_agent_token; %s", token, strings.Join(container.Command, " "))}
```

Comment:
Won't token be shown in `ps` or pod description in this case?

I suggest to use environment variable. Looks like it is already passed, so `echo $SECRET >/usr/yql_agent_token` should be sufficient.